src/main/java/org/shanerx/tradeshop/objects/ShopLocation.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *     Copyright (c) 2016-2017 SparklingComet @ http://shanerx.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * NOTICE: All modifications made by others to the source code belong
+ * to the respective contributor. No contributor should be held liable for
+ * any damages of any kind, whether be material or moral, which were
+ * caused by their contribution(s) to the project. See the full License for more information
+ */
+
+package org.shanerx.tradeshop.objects;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+
+import java.io.Serializable;
+
+public class ShopLocation implements Serializable {
+
+	private transient World world;
+	private String worldName;
+	private double x, y, z;
+
+	public ShopLocation(World w, double x, double y, double z) {
+		this.world = w;
+		this.worldName = w.getName();
+		this.x = x;
+		this.y = y;
+		this.z = z;
+	}
+
+	public ShopLocation(Location loc) {
+		this.world = loc.getWorld();
+		this.worldName = loc.getWorld().getName();
+		this.x = loc.getX();
+		this.y = loc.getY();
+		this.z = loc.getZ();
+	}
+
+	public static ShopLocation deserialize(String loc) {
+		if (loc.startsWith("l")) {
+			String[] locA = loc.split("_"); //Keep same as div
```

Comment:
then why not put `div` in it? instead of keeping the literal.

Reply:
Because that method needs to be static as it needs to be run before you have an object, also div doesn't exist in that file anymore so i just removed the comment

Reply:
I lied, i just put dive back in but either way deserialize returns the object we will need from the string. I may make a constructor that takes the string later but i prefer this as i can return and check for null to check if the string was a location/chunk or not